Cmdline/Action/Replace.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using log4net;
+
+namespace CKAN.CmdLine
+{
+    public class Replace : ICommand
+    {
+        private static readonly ILog log = LogManager.GetLogger(typeof(Replace));
+
+        public IUser User { get; set; }
+
+        public Replace(IUser user)
+        {
+            User = user;
+        }
+
+
+        public int RunCommand(CKAN.KSP ksp, object raw_options)
+        {
+            ReplaceOptions options = (ReplaceOptions) raw_options;
+
+            if (options.ckan_file != null)
+            {                
+                options.modules.Add(LoadCkanFromFile(ksp, options.ckan_file).identifier);
+            }
+
+            if (options.modules.Count == 0 && ! options.replace_all)
+            {
+                // What? No mods specified?
+                User.RaiseMessage("Usage: ckan replace Mod [Mod2, ...]");
+                User.RaiseMessage("  or   ckan replace --all");
+                return Exit.BADOPT;
+            }
+
+            User.RaiseMessage("\r\nReplacing modules...\r\n");
```

Comment:
Currently this message is printed even if nothing is replaced:

```
$ mono ckan.exe replace --all

Replacing modules...


Done!
```
... which is a little misleading as no modules have actually been replaced.
Let's move this to right before the `GetInstance.Replace` call, and check `.Count>0` first. Then we can tell the user something like, "No replaceable mods found" if that's the case.

Reply:
I just tried this with a validly replaceable mod, and the output is the same. We should also print a message for each mod that is successfully replaced.

Reply:
There should probably also be a Y/n confirmation prompt. When installing, we list the mods to be installed and prompt the user to confirm, even though they have just typed those exact names at the command line:

```
About to install...

 * Astrogator v0.7.8(cached)

Continue? [Y/n] 
```

But for replacement, when I have not even typed the name of the mod to install, it just does it silently.

Reply:
Upgrade prints a similar message before anything is actually started or tested.